API/projectClass/customFileClass.m

```diff
@@ -319,9 +320,31 @@ function displayTable(obj)
                 throw(exceptions.invalidPath(sprintf('The given path (%s) is not a valid directory', path)));
             end
         end
+        
+        function filename = addFileExtension(filename, language)
+            % Adds a file extension to the matlab and python filename if it is
+            % missing.
+            %
+            % fileWithExt = obj.addFileExtension('file', 'python')
+            if isempty(filename)
+                return
+            end
+            
+            [~, ~, ext] = fileparts(filename);
+            if isempty(ext)
+                if strcmp(language, supportedLanguages.Matlab.value)
+                    filename = [filename, '.m'];
+                elseif strcmp(language, supportedLanguages.Python.value)
+                    filename = [filename, '.py'];
+                end
```

Comment:
Do we want to raise a warning/error if language is C++ and extension is empty?

Reply:
Dylibs are binary files so extension doesn't matter, I will leave as is for now